File: pi_MC.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void){
  int niter=1000000;
  srand((unsigned)time(NULL)); 

  int n_in=0;
  /*************/
  /* Main loop */
  /*************/
  for(int iter=1;iter<niter+1;iter++){

    double x = (double)rand()/RAND_MAX;
    double y = (double)rand()/RAND_MAX;

    if(x*x+y*y < 1e0)
      n_in=n_in+1;
    printf("%d   %.10f\n",iter,(double)n_in/iter);}
}

File: pi_MC.py
import numpy as np
# seed of random numbers is the system time by default

NITER = 100000
n_in = 0
#################
### Main Loop ###
#################
for iter in range(NITER):
    x = np.random.uniform(0, 1)
    y = np.random.uniform(0, 1)
    if(x*x+y*y < 1.0):
        #n_in = n_in + 1
        n_in += 1
    print(iter+1,n_in/(iter+1))


